### PR TITLE
Added new resource to approve/reject  the provider  directlink gateway request 

### PR DESCRIPTION
**Pull request by sushmitha1506**
### Description of the Issue:
Currently, there is no  IBM terraform resource to accept/reject the connection requests coming from terraform providers.
The current flow is,  Once after provide calls it's API to create/update/delete the direct link gateway respected IBM customer will receive the notification to approve/reject the requests. Respected IBM customer based on his need he will accept/reject the connection request via the portal. 

**PacketFabric** is one of the providers that the customer is using to use the IBM Cloud.
PacketFabric portal : https://docs.packetfabric.com/cloud/ibm/create/
Packet Fabric has a requirement to automate the terraform with respect to the customer portal.
below is the issue raised by the provider
**Issue**: https://github.com/IBM-Cloud/terraform-provider-ibm/issues/3978 

### Description of the Enhancement:
Provide the new  IBM Terraform resources to   Accept/Reject the provider  API  calls to create a gateway, update the gateway, and delete the gateway in the IBM cloud platform.

####  Below are the use cases to which  Terraform resources need to be added to the IBM Terraform to meet the packetfabric requirements.

Terraform Apply (new):
1. user creates the Direct Link GW through the Partner (example with PacketFabric)
Review and accept/reject the IBM Direct Link connection request automatically.

2. Terraform Apply (update):
user upgrades the Direct Link GW through the Partner (example with PacketFabric)
Review and accept/reject the IBM Direct Link connection request automatically.

3. Terraform Destroy (delete):
user deletes the Direct Link GW through the Partner (example with PacketFabric)
Review and accept/reject the IBM Direct Link connection request automatically.

So Terraform when using the proposed new **ibm_dl_gateway_accept** resource would need to handle accepting/reject each change coming from the provider (create/update/delete).

### CODE REVIEW

File: website/docs/r/dl_gateway_action.html.markdown [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+	bgp_asn =  64999
+	name = "example-Gateway"
+	speed_mbps = 1000 
+	port = data.ibm_dl_provider_ports.test_ds_dl_ports.ports[0].port_id
+	customer_account_id = "3b1bd7fa2bc3406ea70ba4ade8aa3f1b"			  		                  
+}
+data "ibm_dl_gateway" "test_ibm_dl_gateway" {
+	name = ibm_dl_provider_gateway.test_dl_gateway.name		
+}
+resource ibm_dl_gateway_action "test_dl_gateway_action" {
```
Comment (rjouhann) [inline website/docs/r/dl_gateway_action.html.markdown:32]: The quotes are missing here "ibm_dl_gateway_action"

File: website/docs/r/dl_gateway_action.html.markdown [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+- `bfd_multiplier` - (String) The number of hello packets not received by a neighbor that causes the originating interface to be declared down.Applicable only for create_gateway_approve requests.
+- `connection_mode` - (Optional, String) Type of network connection that you want to bind to your direct link. Allowed values are `direct` and `transit`.Applicable only for create_gateway_approve requests.
+- `global`- (Required,Bool) Required-Gateway with global routing as **true** can connect networks outside your associated region.Applicable only for create_gateway_approve requests.
+- `metered`- (Required, Bool) Metered billing option. If set **true** gateway usage is billed per GB. Otherwise, flat rate is charged for the gateway.Applicable only for create_gateway_approve requests.
+- `resource_group` - (Optional, Forces new resource, String) The resource group. If unspecified, the account's default resource group is used.Applicable only for create_gateway_approve requests.
+- `default_export_route_filter` - (String) The default directional route filter action  that applies to routes that do not match any directional route filters.Applicable only for create_gateway_approve requests.
```
Comment (rjouhann) [inline website/docs/r/dl_gateway_action.html.markdown:129]: is this a required or optional attribute?

File: website/docs/r/dl_gateway_action.html.markdown [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+- `default_import_route_filter` - (String) The default directional route filter action  that applies to routes that do not match any directional route filters.Applicable only for create_gateway_approve requests.
```
Comment (rjouhann) [inline website/docs/r/dl_gateway_action.html.markdown:130]: is this a required or optional attribute?

File: website/docs/r/dl_gateway_action.html.markdown [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+- `bfd_multiplier` - (String) The number of hello packets not received by a neighbor that causes the originating interface to be declared down.Applicable only for create_gateway_approve requests.
+- `connection_mode` - (Optional, String) Type of network connection that you want to bind to your direct link. Allowed values are `direct` and `transit`.Applicable only for create_gateway_approve requests.
+- `global`- (Required,Bool) Required-Gateway with global routing as **true** can connect networks outside your associated region.Applicable only for create_gateway_approve requests.
+- `metered`- (Required, Bool) Metered billing option. If set **true** gateway usage is billed per GB. Otherwise, flat rate is charged for the gateway.Applicable only for create_gateway_approve requests.
+- `resource_group` - (Optional, Forces new resource, String) The resource group. If unspecified, the account's default resource group is used.Applicable only for create_gateway_approve requests.
```
Comment (rjouhann) [inline website/docs/r/dl_gateway_action.html.markdown:128]: let's mention id in the description and make it required

**Comment by hkantare** [comment]
Fix all the travis failures

File: website/docs/r/dl_gateway_action.html.markdown (hunk 1 above)
Comment (sushmitha1506) [reply website/docs/r/dl_gateway_action.html.markdown:32]: Fixed

File: website/docs/r/dl_gateway_action.html.markdown (hunk 3 above)
Comment (sushmitha1506) [reply website/docs/r/dl_gateway_action.html.markdown:130]: Fixed

**Pull request closed without merging**